<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Methods for estimation of genetic parameters | QG Notes</title>
  <meta name="description" content="These notes are to be used on our QG Analytics website." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Methods for estimation of genetic parameters | QG Notes" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="These notes are to be used on our QG Analytics website." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Methods for estimation of genetic parameters | QG Notes" />
  
  <meta name="twitter:description" content="These notes are to be used on our QG Analytics website." />
  

<meta name="author" content="Peter Sørensen" />


<meta name="date" content="2022-02-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="estimation-of-genetic-parameters.html"/>
<link rel="next" href="applications.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html"><i class="fa fa-check"></i><b>3</b> Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#infinitesimal-model"><i class="fa fa-check"></i><b>3.1</b> Infinitesimal model</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#genetic-effects-genetic-values-and-breeding-values"><i class="fa fa-check"></i><b>3.1.1</b> Genetic effects, genetic values and breeding values</a></li>
<li class="chapter" data-level="3.1.2" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#infinite-number-of-loci-each-with-small-effect-on-the-phenotype"><i class="fa fa-check"></i><b>3.1.2</b> Infinite number of loci each with small effect on the phenotype</a></li>
<li class="chapter" data-level="3.1.3" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#genetic-parameters"><i class="fa fa-check"></i><b>3.1.3</b> Genetic parameters</a></li>
<li class="chapter" data-level="3.1.4" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#basic-questions-remain"><i class="fa fa-check"></i><b>3.1.4</b> Basic questions remain</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="estimation-of-genetic-parameters.html"><a href="estimation-of-genetic-parameters.html"><i class="fa fa-check"></i><b>4</b> Estimation of Genetic Parameters</a>
<ul>
<li class="chapter" data-level="4.1" data-path="estimation-of-genetic-parameters.html"><a href="estimation-of-genetic-parameters.html#genetic-model"><i class="fa fa-check"></i><b>4.1</b> Genetic model</a></li>
<li class="chapter" data-level="4.2" data-path="quantitative-genetics.html"><a href="quantitative-genetics.html#genetic-parameters"><i class="fa fa-check"></i><b>4.2</b> Genetic parameters</a></li>
<li class="chapter" data-level="4.3" data-path="estimation-of-genetic-parameters.html"><a href="estimation-of-genetic-parameters.html#data-required-for-estimating-genetic-parameters"><i class="fa fa-check"></i><b>4.3</b> Data required for estimating genetic parameters</a></li>
<li class="chapter" data-level="4.4" data-path="estimation-of-genetic-parameters.html"><a href="estimation-of-genetic-parameters.html#statistical-models-and-variance-components"><i class="fa fa-check"></i><b>4.4</b> Statistical models and variance components</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html"><i class="fa fa-check"></i><b>5</b> Methods for estimation of genetic parameters</a>
<ul>
<li class="chapter" data-level="5.1" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#estimating-heritability-using-parent---offspring-regression"><i class="fa fa-check"></i><b>5.1</b> Estimating heritability using parent - offspring regression</a></li>
<li class="chapter" data-level="5.2" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#estimating-heritability-using-anova-for-family-data"><i class="fa fa-check"></i><b>5.2</b> Estimating heritability using ANOVA for family data</a></li>
<li class="chapter" data-level="5.3" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#estimating-heritability-using-restricted-maximum-likelihood"><i class="fa fa-check"></i><b>5.3</b> Estimating heritability using Restricted Maximum Likelihood</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#linear-mixed-model"><i class="fa fa-check"></i><b>5.3.1</b> Linear mixed model:</a></li>
<li class="chapter" data-level="5.3.2" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#likelihood-approach-for-estimating-variance-components"><i class="fa fa-check"></i><b>5.3.2</b> Likelihood approach for estimating variance components</a></li>
<li class="chapter" data-level="5.3.3" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#advantages-of-using-reml-for-estimating-genetic-parameters"><i class="fa fa-check"></i><b>5.3.3</b> Advantages of using REML for estimating genetic parameters</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="methods-for-estimation-of-genetic-parameters.html"><a href="methods-for-estimation-of-genetic-parameters.html#when-to-estimate-variance-components"><i class="fa fa-check"></i><b>5.4</b> When to estimate variance components?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>6</b> Applications</a>
<ul>
<li class="chapter" data-level="6.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>6.1</b> Example one</a></li>
<li class="chapter" data-level="6.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>6.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>7</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">QG Notes</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="methods-for-estimation-of-genetic-parameters" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Methods for estimation of genetic parameters</h1>
<p>In general estimation of heritability and genetic correlation is based on methods that determine resemblance
between genetically related individuals. Here we will present three methods for estimating heritability, parent-offspring regression, analysis of variance (ANOVA) for family data (e.g. halfsib/fullsib families) and restricted maximum likelihood (REML) analysis for general pedigree.</p>
<div id="estimating-heritability-using-parent---offspring-regression" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Estimating heritability using parent - offspring regression</h2>
<p>The simplest method for estimation genetic parameters is based on regression analysis.
Heritability may be estimated by comparing phenotypes for traits recorded in parent and offspring. Parent-offspring regression compares trait values in parents (<span class="math inline">\(y_p\)</span>) to trait values in their offspring (<span class="math inline">\(y_o\)</span>). Estimation of heritability is based on a linear regression model:
<span class="math display">\[\begin{align}
            y_o &amp;= y_p b_{o|p}+e_o. \notag \\
\end{align}\]</span>
The slope of the regression line (<span class="math inline">\(b_{o|p}\)</span>) approximates the heritability of the trait when offspring values are regressed against the average trait in the parents. If only one parent’s value is used then heritability is twice the slope. In other words the expected value of the regression line is <span class="math inline">\(b_{o|p} = 0.5h^2\)</span> (or h2 when
regression is on mid-parent mean).</p>
<p>To better understand this relationship consider a situation
where we have collected phenotypes on a number of father-offspring families.
From standard regression theory the slope can be determined as:
<span class="math display">\[\begin{align}
            b_{o|f} &amp;= \frac{Cov(y_f,y_o)}{Var(y_o)} \notag \\
\end{align}\]</span>
where <span class="math inline">\(Cov(y_f,y_o)\)</span> is the covariance between the phenotypes of the father and the offspring and <span class="math inline">\(Var(y_o)\)</span> is the variance of the offspring phenotypes.</p>
<p>The phenotypes for the father (<span class="math inline">\(y_f\)</span>) and the offspring (<span class="math inline">\(y_o\)</span>) can be expressed as:
<span class="math display">\[\begin{align}
            y_f &amp;= \mu+a_f+e_f \notag \\
            y_o &amp;= \mu+0.5a_m+0.5a_f+a_{mendelian}+e_o \notag
\end{align}\]</span>
where <span class="math inline">\(\mu\)</span> is the population mean, <span class="math inline">\(a_m\)</span> and <span class="math inline">\(a_f\)</span> are the additive genetic effect for the mother and the father, a_{mendelian} is the mendelian deviation in the offspring, and <span class="math inline">\(e_f\)</span> and <span class="math inline">\(e_o\)</span> are the residual effect for the father and the offspring.</p>
<p>The offspring get half of the genes from each parent and therefore the breeding value for the offspring is the average of the parents’ breeding values plus the Mendelian deviation:
<span class="math display">\[\begin{align}
        a_{\text{offspring}}=\frac{1}{2}a_{\text{father}}+\frac{1}{2}a_{\text{mother}}+a_{\text{mendelian}} \notag  
\end{align}\]</span>
(a = additive genetic value = breeding value)
The term <span class="math inline">\(a_{mendelian}\)</span> is necessary, because two fullsibs <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> both having parents <span class="math inline">\(father\)</span> and <span class="math inline">\(mother\)</span> receive different random samples of the set of parental alleles. Hence the breeding values <span class="math inline">\(a_i\)</span> and <span class="math inline">\(a_j\)</span> of halfsibs <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are not going to be the same.
Furthermore we assume that the breeding values are normally distributed:
<span class="math display">\[\begin{align}
a_{father} \sim N(0,\sigma^2_{a}) \notag \\
a_{mother} \sim N(0,\sigma^2_{a}) \notag \\
a_{mendelian} \sim N(0,0.5\sigma^2_{a}) \notag \\
\end{align}\]</span></p>
<p>Therefor an expression for the covariance between the phenotypes of the parent and the offspring can be derived as:<br />
<span class="math display">\[\begin{align}
            Cov(y_f,y_o) &amp;= Cov(a_f+e_f,0.5a_m+0.5a_f+a_{mendelian}+e_o) \notag \\
                         &amp;= Cov(a_f,0.5a_f) \notag \\
                         &amp;= 0.5Cov(a_f,a_f) \notag \\
                         &amp;= 0.5\sigma_a^2 \notag 
\end{align}\]</span>
and similar for the variance variance of the offspring phenotypes:<br />
<span class="math display">\[\begin{align}
            Var(y_o) &amp;= Var(0.5a_m+0.5a_f+0.5a_{mendelian}+e_o) \notag \\
                     &amp;= Var(0.5a_m) + Var(0.5a_f) + Var(a_{mendelian}) + Var(e_o) \notag \\
                     &amp;= 0.25Var(a_m) + 0.25Var(a_f) + Var(a_{mendelian}) + Var(e_o) \notag \\
                     &amp;= 0.25\sigma_a^2 + 0.25\sigma_a^2 + 0.5\sigma_a^2 + \sigma_e^2 \notag \\
                     &amp;= \sigma_a^2 + \sigma_e^2 \notag
\end{align}\]</span>
Therefore the expected value of the regression coefficient for a father-offspring analysis is:
<span class="math display">\[\begin{align}
            b_{o|f} &amp;= \frac{0.5\sigma_a^2}{\sigma_a^2 + \sigma_e^2} \notag \\
                    &amp;= 0.5\frac{\sigma_a^2}{\sigma_a^2 + \sigma_e^2} \notag \\
                    &amp;= 0.5h^2 \notag \\
            h^2     &amp;= 2b_{o|f}\notag
\end{align}\]</span></p>
<p>Similar relationships can be derived for other types of parent-offspring regression analysis (mother-offspring and mid- parent-offspring). The heritability can be estimated from the regression coefficient based on:<br />
<span class="math display">\[\begin{align}
    h^2 &amp;= 2b_{o|m} \quad \text{(mother-offspring regression)} \notag \\
    h^2 &amp;= 2b_{o|f} \quad \text{(father-offspring regression)} \notag \\
    h^2 &amp;=  b_{o|mf} \quad \text{(mean parent-offspring regression)} \notag
\end{align}\]</span></p>
<p>Offspring-parent regression is not often used in practice.
It requires data on 2 generations, and uses only this data. It is based on the genetic relationship between parent and offspring which equals 0.5 (i.e. offspring get half of the genes form its parent), but it is not possible to
utilize genetic relationships among parents. However, the method is robust against
selection of parents.</p>
</div>
<div id="estimating-heritability-using-anova-for-family-data" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Estimating heritability using ANOVA for family data</h2>
<p>Genetic parameters have been estimated for many years using analysis of variance
(ANOVA). This method require that individuals can be assigned to groups with the
same degree of genetic relationship for all members. Family structures considered most often
are paternal half-sib groups or full-sib groups. In the case of paternal half-sib group all
offspring of one sire are treated as one group and offspring of different sires are
allocated to different groups.</p>
<p>Estimation of heritability using ANOVA is based on a linear model.
Consider a situation where we have phenotypic observation for multiple offspring for a number of families (halfsib or fullsib).
A simple linear model for the phenotypic observation for the jth offspring in the ith famility include the population mean (<span class="math inline">\(\mu\)</span>) and a family effect (<span class="math inline">\(f_i\)</span>):
<span class="math display">\[\begin{align}
 y_{ij} &amp;= \mu + f_i + e_{ij}
\end{align}\]</span></p>
<p>Assume n observations, with <span class="math inline">\(n_f\)</span> families, with <span class="math inline">\(n_o\)</span> is the number of offspring per family.</p>
<p>The total sum of squares (SST) is the sum of each of the observations squared:
<span class="math display">\[\begin{align}
 SST &amp;= \sum_{i=1}^{n_f}\sum_{j=1}^{n_o} {y}_{ij}^2
\end{align}\]</span></p>
<p>where <span class="math inline">\(y_{ij}\)</span> is an observation on the jth offspring in the ith family.</p>
<p>The mean sum of squares (SSM) is n times the mean squared:
<span class="math display">\[\begin{align}
 SSM &amp;= n\bar{y}_{..}^2
\end{align}\]</span></p>
<p>The model sum of squares (SSA) due to a particular factor (e.g. the family effect) is therefore the sum over all observations of the estimated (family) effect in each observation squared (in balanced data this
is the difference between the family group mean and the overall mean):
<span class="math display">\[\begin{align}
 SSA &amp;= \sum_{i=1}^{n_f} (\bar{y}_{i.}-\bar{y}_{-.})^2
\end{align}\]</span>
Notice that the sum of squares for the main effect (SSA) is the sum of all the squared
estimates of <span class="math inline">\(f_i\)</span>, because in a balanced data set the estimate of <span class="math inline">\(f_i\)</span> is equal to (<span class="math inline">\(y_{i}.-y_{..}\)</span>). In a
balanced data, it is rather simple to determine the expectations for each sum of squares,
because the number of observations per class of f is constant (<span class="math inline">\(n_o\)</span>).</p>
<p>The residual sum of squares (SSE) due to the residual (error) is the sum over all
observations of the residual effect in each observation squared (this is the
difference between the observation and its group mean):</p>
<p><span class="math display">\[\begin{align}
 SSE &amp;= \sum_{i=1}^{n_f}\sum_{j=1}^{n_o} {y}_{ij}^2
\end{align}\]</span></p>
<p>The total sum of squares can be expressed as a sum of the components described above:<br />
<span class="math display">\[\begin{align}
 SST &amp;= SSM + SSA + SSE
\end{align}\]</span></p>
<p>In balanced data, it is rather simple to estimate variance components, by setting
the “Mean Squares” equal to their expectations. Those expectations are linear functions
of the variance components and is derived based on statistical theory (Expected value
of a sum of squares).
In this simple model we can calculate estimate of the residual variance components (<span class="math inline">\(\hat{\sigma}^2_{e}\)</span>) as:
<span class="math display">\[\begin{align}
 \hat{\sigma}^2_{e} &amp;= SSE/(n-n_f)
\end{align}\]</span>
and and estimate of the family variance (<span class="math inline">\(\hat{\sigma}^2_{f}\)</span>) as:
<span class="math display">\[\begin{align}
 \hat{\sigma}^2_{f} &amp;= (SSA/(n_f-1) - \hat{\sigma}^2_{e})/n_o
\end{align}\]</span></p>
<p>Calculating the variance between groups, involves partitioning the sum of squared
observations (SS) due to different sources of variation in the model of analysis, groups
of relatives being one of them, and equating the corresponding mean squares. Mean
squares are derived as the SS divided by the associated degrees of freedom, to their
expectations.</p>
<p>Using ANOVA, the covariance among members of a family or group of relatives is
usually determined as the variance component between groups. For example, in case of
a half-sib family model, the variance between half-sib families <span class="math inline">\(\sigma_{s}^2\)</span>
and variance within half-sib families <span class="math inline">\(\sigma_{e}^2\)</span>.
As shown earlier, the half-sib family variance <span class="math inline">\(\sigma_{s}^2=0.25\sigma_{a}^2\)</span>
while the variance within half-sib families is <span class="math inline">\(0.75\sigma_{a}^2+\sigma_{e}^2\)</span></p>
<p>Fullsib families
<span class="math display">\[\begin{align}
            y_m &amp;= \mu+a_m+e_m \notag \\
            y_f &amp;= \mu+a_f+e_f \notag \\
            y_{o1} &amp;= \mu+0.5a_m+0.5a_f+0.5a_{mendelian_1}+e_{o1} \notag \\
            y_{o2} &amp;= \mu+0.5a_m+0.5a_f+0.5a_{mendelian_2}+e_{o2} \notag
\end{align}\]</span>
<span class="math display">\[\begin{align}
            Cov(y_{o1},y_{o2}) &amp;= Cov(0.5a_m+0.5a_f+a_{mendelian1}+e_{o1},0.5a_m+0.5a_f+a_{mendelian2}+e_{o2}) \notag \\
                         &amp;= Cov(0.5a_f,0.5a_f) + Cov(0.5a_m,0.5a_m) \notag \\
                         &amp;= 0.25Cov(a_f,a_f)+0.25Cov(a_m,a_m) \notag \\
                         &amp;= 0.25\sigma_a^2 +0.25\sigma_a^2 \notag \\
                         &amp;= 0.5\sigma_a^2 \notag
\end{align}\]</span>
Halfsib families
<span class="math display">\[\begin{align}
            y_{m1} &amp;= \mu+a_{m1}+e_{m1} \notag \\
            y_{m2} &amp;= \mu+a_{m2}+e_{m2} \notag \\
            y_f &amp;= \mu+a_f+e_f \notag \\
            y_{o1} &amp;= \mu+0.5a_{m1}+0.5a_f+0.5a_{mendelian1}+e_{o1} \notag \\
            y_{o2} &amp;= \mu+0.5a_{m2}+0.5a_f+0.5a_{mendelian2}+e_{o2} \notag
\end{align}\]</span>
<span class="math display">\[\begin{align}
            Cov(y_{o1},y_{o2}) &amp;= Cov(0.5a_{m1}+0.5a_f+a_{mendelian1}+e_{o1},0.5a_{m2}+0.5a_f+a_{mendelian2}+e_{o2}) \notag \\
                         &amp;= Cov(0.5a_f,0.5a_f) + Cov(0.5a_{m1},0.5a_{m2}) \notag \\
                         &amp;= 0.25Cov(a_f,a_f) + 0 \notag \\
                         &amp;= 0.25\sigma_a^2 \notag \\
                         &amp;= 0.25\sigma_a^2 \notag
\end{align}\]</span></p>
<p>Therefore the heritability can be estimated from the variance component based on:<br />
<span class="math display">\[\begin{align}
    \sigma_{hs}^2 &amp;= 0.25\sigma_a^2 \quad \text{(halfsib families)} \notag \\
    \sigma_{fs}^2 &amp;= 0.5\sigma_a^2 \quad \text{(halfsib families)} \notag \\
    h^2 &amp;= \frac{4\sigma_{hs}^2}{4\sigma_{hs}^2+\sigma_e^2} \quad \text{(halfsib families)} \notag \\
    h^2 &amp;= \frac{2\sigma_{fs}^2}{2\sigma_{fs}^2+\sigma_e^2} \quad \text{(fullsib families)} \notag \\
\end{align}\]</span></p>
<p>Data arising from experimental designs used for estimating genetic parameters
are usually not balanced (i.e. number of offspring varies across families).
Methods analogous to the ANOVA have been developed for unbalanced data.
However REML is nowadays the method of choice for variance component estimation.</p>
<p>The estimation of variance components (within and between family components). If
the variation within families is large relative to differences between families, the trait
must be lowly heritable. Variance components are attributed to specific effects. For
example, the (paternal) half-sib variance is due to differences between sires. The
variance component represents the sire variance, which is a quarter of the additive
genetic variance.</p>
<p>Estimation of variance components is easier to generalise, and this method is generally
used to estimate genetic parameters. The remainder of this chapter will therefore mostly deal with
variance component estimation.</p>
</div>
<div id="estimating-heritability-using-restricted-maximum-likelihood" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Estimating heritability using Restricted Maximum Likelihood</h2>
<p>Genetic parameters are nowadays estimated using restricted maximum likelihood (REML) or Bayesian methods.
This method allow for estimation of genetic parameters using phenotypic information for individuals from a general pedigree.REML is based on linear mixed model methodology and use a likelyhood approach for estimating genetic parameters.</p>
<div id="linear-mixed-model" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Linear mixed model:</h3>
<p>The linear mixed model contains the observation vector for the trait(s) of interest, the
factors that explain how the observations came to be, and a residual effect that includes
everything not explainable.</p>
<p>A matrix formulation of a general model equation is:
<span class="math display">\[\begin{align}
y &amp;= Xb + Za + e \notag
\end{align}\]</span></p>
<p>where
<span class="math display">\[\begin{align}
y &amp;: \text{is the vector of observed values of the trait,} \notag \\
b &amp;: \text{is a vector of factors, collectively known as fixed effects,} \notag \\
a &amp;: \text{is a vector of factors known as random effects,} \notag \\
e &amp;: \text{is a vector of residual terms, also random,} \notag \\
X,Z &amp;: \text{are known design matrices that relate the elements of b and a to their corresponding element in y.} \notag 
\end{align}\]</span></p>
<p>The <strong>observation vector</strong> contains elements resulting from measurements, either subjective
or objective, on the experimental units (usually animals) under study. The elements in
the observation vector are random variables that have a multivariate distribution, and if
the form of the distribution is known, then advantage should be taken of that knowledge.
Usually <span class="math inline">\(y\)</span> is assumed to have a multivariate normal distribution, but that is not always
true. The elements of <span class="math inline">\(y\)</span> should represent random samples of observations from some defined
population. If the elements are not randomly sampled, then bias in the estimates of b and
<span class="math inline">\(a\)</span> can occur, which would lead to errors in ranking individuals.</p>
<p>A <strong>continuous factor</strong> is one that has an infinite-like range of
possible values. For example, if the observation is the distance a rock can be thrown,
then a continuous factor would be the weight of the rock. If the observation is the
rate of growth, then a continuous factor would be the amount of feed eaten.</p>
<p><strong>Discrete</strong> factors usually have classes or levels such as age at calving might have four
levels (e.g. 20 to 24 months, 25 to 28 months, 29 to 32 months, and 33 months or
greater). An analysis of milk yields of cows would depend on the age levels of the
cows.</p>
<p>In the traditional ”frequentist” approach, <strong>fixed</strong> and <strong>random</strong> factors need
to be distinguished.</p>
<p>If the number of levels of a factor is small or limited to a fixed number, then that
factor is usually fixed.
If inferences about a factor are going to be limited to that set of levels, and to no
others, then that factor is usually fixed.
If a new sample of observations were made (a new experiment), and the same levels
of a factor are in both samples, then the factor is usually fixed.
If the levels of a factor were determined as a result of selection among possible
available levels, then that factor should probably be a fixed factor.
Regressions of a continuous factor are usually a fixed factor (but not always).</p>
<p>If the number of levels of a factor is large, then that factor can be a
random factor.
If the inferences about a factor are going to be made to an entire population of
conceptual levels, then that factor can be a random factor.
If the levels of a factor are a sample from an infinitely large population, then that
factor is usually random.
If a new sample of observations were made (a new experiment), and the levels were
completely different between the two samples, then the factors if usually random.</p>
<div id="expectation-and-variance-of-variables-in-the-model" class="section level4 unnumbered">
<h4>Expectation and variance of variables in the model:</h4>
<p>In the statistical model (specified above) the random effect (<span class="math inline">\(a\)</span> and <span class="math inline">\(e\)</span>) and the phenotypes (<span class="math inline">\(y\)</span>) are considered to be random variables that are assumed to follow a multivariate normal distribution.<br />
In general terms the expectations of these random variables are:<br />
<span class="math display">\[\begin{align}
E(y) &amp;= E(Xb) + E(Za) + E(e) \notag \\
     &amp;= Xb + 0 + 0 \notag \\
     &amp;= Xb \notag
\end{align}\]</span>
and the variance-covariance matrices are:
<span class="math display">\[\begin{align}
 Var(a) &amp;= G \notag \\
 Var(e) &amp;= R \notag \\
 Var(y) &amp;= ZGZ&#39; + R = V \notag \\
\end{align}\]</span></p>
<p>where <span class="math inline">\(G\)</span>, <span class="math inline">\(R\)</span> and <span class="math inline">\(V\)</span> are square matrices of genetic, residual and phenotypic (co)variances among the individuals in the data set. If we assume</p>
<p><span class="math display">\[\begin{align}
 Var(a) &amp;= G \notag \\
        &amp;= A\sigma_a^2 \notag \\
 Var(e) &amp;= R \notag \\
        &amp;= I\sigma_e^2 \notag \\
 Var(y) &amp;= ZGZ&#39; + R = V \notag \\
        &amp;= A\sigma_a^2 + I\sigma_e^2 \notag
\end{align}\]</span></p>
</div>
<div id="assumptions-and-limitations-of-the-model" class="section level4 unnumbered">
<h4>Assumptions and limitations of the model:</h4>
<p>The third part of a model includes items that are not apparent in parts 1 and 2. For
example, information about the manner in which data were sampled or collected. Were
the animals randomly selected or did they have to meet some minimum standards? Did
the data arise from many environments, at random, or were the environments specially
chosen? Examples will follow. A linear model is not complete unless all three parts of the model are present.
Statistical procedures and strategies for data analysis are determined only after a complete
model is in place.</p>
</div>
</div>
<div id="likelihood-approach-for-estimating-variance-components" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Likelihood approach for estimating variance components</h3>
<p>Restricted Maximum Likelihood is a method that is used to estimate the parameters in the model (i.e. variance components <span class="math inline">\(\sigma_{a}^2\)</span> and <span class="math inline">\(\sigma_{e}^2\)</span>) specified in the linear mixed model above. The general principle used in likelihood methods is to find the set of parameters which maximizes the likelihood of the data.</p>
<p>It is useful to recall that the likelihood (<span class="math inline">\(L(\theta|{y})\)</span>) is any function of the parameter (<span class="math inline">\(\theta\)</span>) that is proportional to <span class="math inline">\(p({y}|\theta)\)</span>. Maximizing <span class="math inline">\(L(\theta|{y})\)</span> leads to obtaining the most likely value of <span class="math inline">\(\theta\)</span> (<span class="math inline">\(\hat{\theta}\)</span>) given the data <span class="math inline">\({y}\)</span>. Usually the likelihood is expressed in terms of its logarithm (<span class="math inline">\(l(\theta|\bf{y})\)</span>) as it makes the algebra easier.</p>
<p>The likelihood of the data for a given linear mixed model can be written as a function;</p>
<p><span class="math display">\[\begin{equation} 
    l(\matr{V}|\vect{y}, \matr{X}, \bfbeta) 
    \propto
     -\frac{1}{2}\ln|\matr{V}|
     -\frac{1}{2}\ln|\matr{X&#39;V}^{-1}\matr{X}|
     -\frac{1}{2}(\vect{y}-\matr{X}{\bfbeta})&#39;\matr{V}^{-1}(\vect{y}-\matr{X}{\bfbeta})
    \label{eq:reml.likelihood}
\end{equation}\]</span></p>
<p>From calculus we know that we can find the maximum of a function by taking the first
derivative and set that equal to zero. Solving that would result in the desired
parameters (assuming that we did not find the minimum, this can be checked using
second derivatives). The first and second derivatives of the likelihood function
are complicated formulas.</p>
<p>There are no simple one-step solutions for estimating the variance components based on REML . Instead, we infer the partial derivatives of the likelihoods with respect to the variance components. The solutions to these involve the inverse of the variance-covariance matrix, which themselves includes the variance components, so the variance components estimates are non-linear functions of the variance components. It is therefore necessary to apply iterative methods to obtain the estimates.</p>
<p>From the estimate of the variance components the heritability can easily computed by
<span class="math display">\[\begin{align}
\hat{h}^2 &amp;= \hat{\sigma}^2_{a}/(\hat{\sigma}^2_a+\hat{\sigma}^2_e)
\end{align}\]</span>
where the “<span class="math inline">\(\hat{h}^2\)</span>” refers to the heritability is an estimate.</p>
</div>
<div id="advantages-of-using-reml-for-estimating-genetic-parameters" class="section level3" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> Advantages of using REML for estimating genetic parameters</h3>
<p>The REML method was developed by  as an improvement of the standard Maximum Likelihood (ML). The ML method was originally proposed by  but was introduced to variance components estimation by . ML assumes that fixed effects are known without error which is in most cases false and, as consequence, it produces biased estimates of variance components (usually, the residual variance is biased downward). To solve this problem, REML estimators maximize only the part of the likelihood not depending on the fixed effects,
by assuming that the fixed effects have been, so to speak, fixed. This entails that when comparing multiple models by their REML likelihoods, they must contain the same fixed effects,
and that REML, by itself, does not estimate the fixed effects.</p>
<p>REML does not produce unbiased estimates owing to the inability to return negative values of variance components of many methods to obtain REML estimators, but it is still the method of choice due to the fact that this source of bias is also present in ML estimates .</p>
<p>REML requires that y have a multivariate normal distribution
although various authors have indicated that ML or REML estimators may be an
appropriate choice even if normality does not hold (Meyer, 1990).</p>
<p>REML can account for selection when the complete mixed model is used with
all genetic relationships and all data used for selection included (Sorensen and Kennedy,
1984; Van der Werf and De Boer, 1990).</p>
<p>There is obviously an advantage in using (RE)ML methods that are more flexible in
handling animal and plant breeding data on several (overlapping) generations (and possibly
several random effects). However, the use of such methods has a danger in the sense
that we need not to think explicitly anymore about data structure. To estimate, as an
example, additive genetic variance, we need to have a data set that contains a certain
family structure that allows us to separate differences between families from differences
within families. Or in other words, we need to separate genetic and residual variance.
ANOVA methods require more explicit knowledge about such structure, since the data
has to be ordered according to family structures (e.g. by half sib groups).</p>
<p>Developments in variance component estimation specific to animal and plant breeding have been
closely linked with advances in the genetic evaluation using Best Linear
Unbiased Prediction (BLUP). Early REML applications were generally limited to models
largely equivalent to those in corresponding ANOVA type analysis, considering one
random effect only and estimating genetic variances from paternal half sib covariances
(so-called sire model). Today, heritability can be estimated from general pedigrees and from genomic relatedness estimated from genetic markers. Linear mixed models is also used genetic evaluation schemes,
allowing information on all known relationships between individuals to be incorporated in
the analysis. Linear mixed models can include maternal, permanent environmental, cytoplasmic or
dominance effects or effects at QTL thereby more accurately describe the observed data.
These effects are fitted as additional random effects.</p>
</div>
</div>
<div id="when-to-estimate-variance-components" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> When to estimate variance components?</h2>
<p>In general, the estimation of variances and covariances has to be based on a sufficient
amount of data. Depending on the data structure and the circumstances during
measuring, estimations can be based on some hundreds (selection experiments) or more
than 10,000 observations (field recorded data). It is obvious that we are not interested in
estimating variance components from every data set. The information in literature is in
many cases even better than estimations based on a small data set. In general, we have
to estimate variance if:
- we are interested in a new trait, from which no parameters are available;
- variances and covariances might have changed over time
- considerable changes have occurred in a population e.g. due to recent
importations.
Mostly it is assumed that variances and covariances, and especially the ratio of both of
them (like heritability, correlation), are based on particular biological rules, which do
not rapidly change over time. However, it is well known that the genetic variance
changes as consequence of selection. Changes are especially expected in situations with
short generation intervals, high selection intensities or high degrees of inbreeding or in a
situation in which a trait is determined by only a few genes. Secondly, the
circumstances under which measurements are taken can change. If conditions are
getting more uniform over time, the environmental variance decreases, and
consequently the heritability increases. Thirdly, the biological interpretation of a trait
can change as consequence of a changed environment; feed intake under limited
feeding is not the same as feed intake under ad-lib feeding. In conclusion, there are
sufficient reasons for regular estimation of (co-)variance components.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="estimation-of-genetic-parameters.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="applications.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["qgnotes.pdf", "qgnotes.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
